Update updating-the-sdk.md (#212)
vrchat-community/creator-docs: Docs/docs/sdk/updating-the-sdk.md

slug: "updating-the-sdk"
hidden: false
createdAt: "2019-07-24T03:33:36.520Z"

> updatedAt: "2024-12-28T00:15:39.000Z"

Reviewer: This line can safely be removed instead of being updated.
```suggestion
```

## Manual Updates
If you're using the VCC, you don't need to worry about manual updates.

> However, if you wish to, you can download the SDK on the [VRChat website](https://vrchat.com/home/download) after signing into your VRChat account

Reviewer: ```suggestion
However, if you wish to, you can download the SDK on the [VRChat website](https://vrchat.com/home/download) after signing into your VRChat account.
```

## Manual Updates
If you're using the VCC, you don't need to worry about manual updates.

> However if you wish to. You can get the sdk at [VRChat home/download](https://vrchat.com/home/download) after signing in.

Reviewer: ```suggestion
However, if you wish to, you can download the SDK on the [VRChat website](https://vrchat.com/home/download) after signing into your VRChat account
```

3. Import the new SDK3 - Avatars package into that project.
4. Close that Unity project.
5. Using Explorer (Do not open Unity yet!), delete the VRCSDK3 folders from the project you're upgrading. Until this guide says otherwise, ***do not open Unity.***

> 6. Edit the references of the vrchat SDK3 from Packages/vpm-manifest.json to the new SDK version from the project you're upgrading.
> 7. From your new blank project that you imported the SDK into, copy the VRCSDK3 folders into your project that you're upgrading.
> 8. Once the copy completes, open Unity and open your upgraded project. You may delete the blank project.

Reviewer: ```suggestion
6. Edit the references of the VRChat SDK3 from `Packages/vpm-manifest.json` to the new SDK version from the project you're upgrading.
7. From your new blank project that you imported the SDK into, copy the VRCSDK3 folders into the project that you're upgrading.
8. After you copy the files, open Unity and open your upgraded project. You may delete the blank project.
```

1. Close Unity
2. Back up project to another folder (do not back up the Library folder, these files are auto-generated by Unity)
3. Delete SDK and Plugins folder, as well as the the associated .META files
4. Edit the references of the VRChat SDK3 from `Packages/vpm-manifest.json` to the new SDK version from the project you're upgrading.
5. Create a new "dummy" Unity Project by creating a blank Unity project with Unity Hub.
6. Install the latest VRChat SDK on the dummy project.

> 7. Copy the newly added `SDK/Plugin` folder and associated `.meta` files from the dummy project into your original project

Reviewer: ```suggestion
7. Copy the newly added `SDK/Plugin` folder and associated `.meta` files from the dummy project into your original project.
```

1. Close Unity
2. Back up project to another folder (do not back up the Library folder, these files are auto-generated by Unity)
3. Delete SDK and Plugins folder, as well as the the associated .META files

> 4. Edit the references of the vrchat SDK3 from Packages/vpm-manifest.json to the new SDK version from the project you're upgrading.
> 5. Create a new dummy Unity Project
> 6. Install the latest SDK on the dummy project
> 7. Copy the newly added SDK/Plugin and associated .META files from the dummy project into your original project
> 8. Done. Now you can open your upgraded project, and there will be no crash no matter how complex your trigger work!

Reviewer: ```suggestion
4. Edit the references of the VRChat SDK3 from `Packages/vpm-manifest.json` to the new SDK version from the project you're upgrading.
5. Create a new "dummy" Unity Project by creating a blank Unity project with Unity Hub.
6. Install the latest VRChat SDK on the dummy project.
7. Copy the newly added `SDK/Plugin` folder and associated `.meta` files from the dummy project into your original project
8. Done. You can now open your upgraded project.
```
